### DISCUSSION TITLE
x-init calls twice if function passed is called init

### DISCUSSION THREAD

**Discussion by davidwebca** [alpinejs/alpine#1799] [7. Bugs (archived)]
Hello! 

I think this is mostly about an upgrade issue from x-init in version 3. As I understand it, the new version will call init() automatically if it's found on a component, but the issue here is that we're not even allowed to use a function called init() manually inside the x-init call. The usage that I had of that before was to pass $watch or other magics inside my init function. 

```
<!-- Does not work, $watch is not passed and init is called twice -->
<div x-init="init($watch)"></div>
```

Now, I don't know if it's something you really want to fix or if you'd instead advise people to use a different name of function like "start($watch)" instead, but if the latter, I'd suggest adding a clear note or warning in the docs to not use init() "manually" in that way and different ways to access $watch if possible.

**Comment by danddanddand** [comment]
In our project we basically don't allow users to implement init function with parameters (our TS definitions will show it as error). We use naming initData and pass in params or options object.

We did pass around the watch magic like you are when we first migrated to v3 but we've switched to using Alpine.effect() and code is a bit cleaner.

**Reply by davidwebca** [reply]
I was searching for an example to avoid passing $watch and using effects earlier. Would you mind sharing?

**Reply by danddanddand** [reply]
https://ryangjchandler.co.uk/posts/how-to-watch-alpine-stores-for-changes

**Reply by davidwebca** [reply]
Love it! Really clean and makes sense. Thanks 👍
